### PR TITLE
Optionally process sys and user config in embedded app

### PR DESCRIPTION
**Pull request by pincorrect**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### What is the purpose of your _pull request_?
- [ ] Bug fix
- [ ] New extractor
- [x] New feature

---
### Description of your _pull request_ and other information

Explanation of your _pull request_ in arbitrary form goes here. Please make sure the description explains the purpose and effect of your _pull request_ and is worded well enough to be understood. Provide as much context and examples as possible.

The option parsing code in options.py provides a very nice feature to have both system and user config files that specify options without needing to enter them on the command line. This feature was not provided for python apps that embed youtube-dl and call main() with an argv list. My modification allows apps that embed youtube-dl to use system and user config files, if desired.

Although embedding apps that do not want this feature could easily disable it by adding '--ignore-config' as one of the supplied argv args, the behavior of existing apps could change unless so modified. To prevent this, I added an additional argument to main(), ignoreConfig, with a default value True. This ensures that existing apps will run without any change in behavior. To use the new feature, you need to specify ignoreConfig=False.

### CODE REVIEW

File: youtube_dl/__init__.py [hunk 1]
```diff
@@ -382,8 +382,6 @@ def parse_retries(retries):
         'external_downloader_args': external_downloader_args,
         'postprocessor_args': postprocessor_args,
         'cn_verification_proxy': opts.cn_verification_proxy,
-        'geo_verification_proxy': opts.geo_verification_proxy,
```
Comment (yan12125) [inline youtube_dl/__init__.py]: Don't you like it?

Comment (pincorrect) [inline youtube_dl/__init__.py]: Sorry, I did not mean to delete that line. Later today when I can get to my dev system, I'll correct that.

Comment (pincorrect) [inline youtube_dl/__init__.py]: Odd. I did not delete that line. I never had it. But it looks like it was added 17 days ago and I only forked yesterday. I will update my branch from the latest in master, then check the diffs carefully to make sure my branch only has the change I intended.

Comment (pincorrect) [inline youtube_dl/__init__.py]: This is all straightened out now.

**Pull request closed without merging**